### PR TITLE
PP-992, PP-1296: Function add_to_resource_group is adding unwanted characters and doesn't validate the file after creating it

### PR DESCRIPTION
**Pull request by Mahalty**
#### Issue-ID:
**[PP-992](https://pbspro.atlassian.net/browse/PP-992)
[PP-1296](https://pbspro.atlassian.net/browse/PP-1296)**

#### Bug/feature Description:
* Function "add_to_resource_group" is adding unwanted characters and doesn't validate the changes after adding entries to resource_group file.

#### Affected Platform(s)
* All

#### Cause / Analysis / Design
* Due to the addition of unwanted characters and not validating  after adding entries to resource_group file, the changes made are not reflected and jobs doesn't run
* Function "add_to_resource_group" is calling function "create_node" internally which internally calls "FairshareNode"
* Parameters passed to "add_to_resource_group" and "create_node"  are the name,id,parent_name and nshares. We are not passing usage there but the "__init__" method in class FairshareNode is passing the usage value as "unknown" by default, that is the reason why resource_group file has the entry "unknown" at the end.

#### Solution Description
* Update the default value for "usage" to "None" in function FairshareNode*
* Add code to validate the resource_group changes

#### Testing logs/output
* [Execution_logs_TestFairshare_Withfix_PP921.txt](https://github.com/PBSPro/pbspro/files/2313999/Execution_logs_TestFairshare_Withfix_PP921.txt)
**Note:** In the above execution logs,test "test_pbsfs" failed due to [PP-1297](https://pbspro.atlassian.net/browse/PP-1297)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/fw/ptl/lib/pbs_testlib.py [hunk 1]
```diff
@@ -12183,7 +12183,7 @@ def parse_resource_group(self, hostname=None, resource_group=None):
         tree.update()
         return tree
 
-    def add_to_resource_group(self, name, id, parent, nshares):
+    def add_to_resource_group(self, name, id, parent, nshares, validate=True):
```
Comment (bhroam) [inline test/fw/ptl/lib/pbs_testlib.py]: id() is a python built-in.  Consider renaming this parameter.

**Comment by Mahalty** [comment]
@bhroam I have addressed your review comments. Please review.

**Latest execution logs:**
[Execution_logs_TestFairshare_Withfix_PP921_latest.txt](https://github.com/PBSPro/pbspro/files/2317612/Execution_logs_TestFairshare_Withfix_PP921_latest.txt)

File: test/fw/ptl/lib/pbs_testlib.py [hunk 2]
```diff
@@ -12183,7 +12183,7 @@ def parse_resource_group(self, hostname=None, resource_group=None):
...
         """
```
Comment (VishrutArora) [inline test/fw/ptl/lib/pbs_testlib.py]: @Mahalty Consider adding/editing 'fairshare_id' and 'validate' to the docstring.

**Comment by Mahalty** [comment]
Thanks @bhroam  and @VishrutArora  for the review.

I have squashed the commits and rebased my branch. Please take this PR forward.

**Pull request merged**